scripts/fast-reboot

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Check root privileges
+if [ "$EUID" -ne 0 ]
+then
+  echo "Please run as root"
+  exit
+fi
+
+# Unload previous kernel if any loaded
+if [ "x`cat /sys/kernel/kexec_loaded`y" = "x1y" ]
+then
+  /sbin/kexec -u
+fi
+
+# Kernel and initrd image
+KERNEL_IMAGE="/vmlinuz"
+INITRD="/initrd.img"
+BOOT_OPTIONS="`cat /proc/cmdline`"
+
+case "$BOOT_OPTIONS" in
+  *fast-reboot*)
+    # it's already there
+    ;;
+  *)
+    BOOT_OPTIONS="$BOOT_OPTIONS fast-reboot"
+    ;;
+esac
+
+# Load kernel into memory
+/sbin/kexec -l "$KERNEL_IMAGE" --initrd="$INITRD" --append="$BOOT_OPTIONS"
+
+# Stop docker container engine. Otherwise will have broken docker storage
+systemctl stop docker.service
+
+# Wait until all buffers synced with disk
+sync
+sleep 1
+sync
+sleep 1
+sync
+sleep 1
```

Comment:
I think the best method is sleeping here. Because 'Dirty pages' parameter indicates only about the Linux page cache usage. But I'd like to be sure the HDD writes all data back to disk from it's own caches. So it's better just to wait.

P.S.: Another method to sync is to remount the partition.